### ISSUE TITLE
LXD importer failing when proxy is active?

### ISSUE THREAD

**Issue by consultit** [OpenNebula/one#3427]
**Description**

I opened this issue on https://forum.opennebula.org/t/lxd-hypervisor-on-red-hat-virtualization-virtual-machines-container-instantiation-failure/7333 and @dann1 suggested to open an issue on github.

Recap:
We have a test environment on our corporate Virtualization platform (Red Hat Virtualization):
- 2 VMs, frontend and host, with ubuntu 18.04
- OpenNebula: 5.8.1
- OpenNebula hypervisor: LXD
- VMs are behind a corporate proxy

**To Reproduce**
Steps to reproduce the behavior.

From sunstone web interface:
1) downloaded 'ubuntu_xenial - LXD'  from MarketPlace
2) Instantiated a container based on 'ubuntu_xenial - LXD' template
3) Status. FAILURE
Note: the same results apply for any apps

**Expected behavior**

3) Status. READY

**Details**

I executed these commands after the download/instantiation attempt :
```
$ oneimage list

  ID USER       GROUP      NAME            DATASTORE     SIZE TYPE PER STAT RVMS
  12 oneadmin   oneadmin   ubuntu_xenial - default      1024M OS    No used    1

$ oneimage show 12

IMAGE 12 INFORMATION                                                            
ID             : 12                  
NAME           : ubuntu_xenial - LXD 
USER           : oneadmin            
GROUP          : oneadmin            
LOCK           : None                
DATASTORE      : default             
TYPE           : OS                  
REGISTER TIME  : 06/11 10:32:43      
PERSISTENT     : No                  
SOURCE         : /var/lib/one//datastores/1/29df886c8038daf60c1c9c2ccb56c55d
PATH           : "lxd://https://uk.images.linuxcontainers.org//images/ubuntu/xenial/amd64/default/./20190610_07:42/rootfs.tar.xz?size=1024&filesystem=ext4&format=raw"
SIZE           : 1024M               
STATE          : used                
RUNNING_VMS    : 1                   

PERMISSIONS                                                                     
OWNER          : um-                 
GROUP          : ---                 
OTHER          : ---                 

IMAGE TEMPLATE                                                                  
DEV_PREFIX="hd"
DRIVER="raw"
FORMAT="raw"
FROM_APP="69"
FROM_APP_MD5="4997da90d6528e69191fd3dffbd5c600"
FROM_APP_NAME="ubuntu_xenial - LXD"

VIRTUAL MACHINES

    ID USER     GROUP    NAME            STAT UCPU    UMEM HOST             TIME
    25 oneadmin oneadmin ubuntu_xenial - fail    0      0K host1        2d 01h38

$ sudo mount /var/lib/one//datastores/1/29df886c8038daf60c1c9c2ccb56c55d /mnt

$ ls -al /mnt/

total 24
drwxr-xr-x  3 oneadmin oneadmin  4096 giu 11 10:37 .
drwxr-xr-x 24 root     root      4096 giu  6 06:47 ..
drwx------  2 root     root     16384 giu 11 10:37 lost+found

```
so it seems an empty image.
The corporate proxy is configured in opennebula in this way:

```
In "/etc/one/oned.conf":
...
MARKET_MAD = [
    EXECUTABLE = "one_market",
    ARGUMENTS  = "-t 15 -m http,s3,one,linuxcontainers --proxy http://10.0.255.2:80"
]
...

In "/etc/one/sunstone-server.conf":
...
:proxy: http://10.0.255.2:80: 
...

```

BTW, proxy allows downloading of images through browser: for example I downloaded  successfully 
```
https://uk.images.linuxcontainers.org//images/ubuntu/xenial/amd64/default/./20190610_07:42/rootfs.tar.xz?size=1024&filesystem=ext4&format=raw
```
bye

**Additional context**

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by dann1** [comment]
I tested a local proxy setup, with both config files updated to use the local squid and it worked fine.

![image](https://user-images.githubusercontent.com/16429804/59901418-cb942b80-93c0-11e9-8006-90dbd37eeb33.png)

**Comment by consultit** [comment]
Hi, thanks to your answer we investigated our proxy logs. 
We got these lines:
```
1561104826.169     39 172.16.40.3 TCP_MISS/301 360 GET http://localhost:2474/service - FIRST_UP_PARENT/10.3.204.167 -
1561104967.732    274 172.16.40.3 TCP_MISS/000 4249 CONNECT images.linuxcontainers.org:443 - FIRST_UP_PARENT/10.3.204.167 -
1561104967.758    874 172.16.40.3 TCP_MISS/000 13147 CONNECT marketplace.opennebula.systems:443 - FIRST_UP_PARENT/10.3.204.167 -
1561104971.850   4117 172.16.40.3 TCP_MISS/000 49682 CONNECT uk.images.linuxcontainers.org:443 - FIRST_UP_PARENT/10.3.204.167 -
```
The first line shows that `localhost:2474` is contacted through the proxy, it seems incorrect. In your arrangement, this works because it seems your proxy executes on the same host, so this error might be hidden. Other lines show that external sites are contacted correctly through proxy. May be there is a `--noproxy` option somewhere? or what else?
Bye

**Comment by dann1** [comment]
> 1561104826.169     39 172.16.40.3 TCP_MISS/301 360 GET http://localhost:2474/service - FIRST_UP_PARENT/10.3.204.167 -

That seems to be a Sunstone process, maybe you can try using the CLI with `onemarketapp`

**Comment by consultit** [comment]
After doing some investigation we believe that the problem is related to `curl` which does not seem to read the `http_proxy` environment variable, that we correctly set in `/etc/environment`. 
Opening a shell for the user oneadmin, `http_proxy` is exported, but executing `onemarketapp` we found (through `netstat`) that the subshell that performs `curl` does not seem to inherit `http_proxy` because it tries to connect directly to public addresses bypassing the proxy. 
We don't know if we are missing some other configuration (as we said we followed [this guide](http://docs.opennebula.org/5.8/deployment/index.html) for the installation.
bye
